Nevermind, I forgot to pass -l when testing. There are still a few failures:
```
**********************************************************************
File "src/sage/src/sage/doctest/test.py", line 241, in sage.doctest.test
Failed example:
    subprocess.call(["sage", "-tp", "1000000", "--timeout=120",  # long time
         "--die_timeout=10", "--optional=sage",
         "--warn-long", "0", "99seconds.rst", "interrupt_diehard.rst"], **kwds2)
Expected:
    Running doctests...
    Doctesting 2 files using 1000000 threads...
    Killing test 99seconds.rst
    Killing test interrupt_diehard.rst
    ----------------------------------------------------------------------
    Doctests interrupted: 0/2 files tested
    ----------------------------------------------------------------------
    ...
    128
Got:
    usage: sage [-h] [-v] [-q] [--simple-prompt] [-V] [-n [{jupyter,jupyterlab}]]
                [-c [COMMAND]]
                [file]
    sage: error: unrecognized arguments: -tp --timeout=120 --die_timeout=10 --optional=sage --warn-long 0 99seconds.rst interrupt_diehard.rst
    2
**********************************************************************
File "src/sage/src/sage/doctest/test.py", line 259, in sage.doctest.test
Failed example:
    pid = int(open(F).read())
Exception raised:
    Traceback (most recent call last):
      File "/usr/lib/python3.13/site-packages/sage/doctest/forker.py", line 730, in _run
        self.compile_and_execute(example, compiler, test.globs)
        ~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      File "/usr/lib/python3.13/site-packages/sage/doctest/forker.py", line 1154, in compile_and_execute
        exec(compiled, globs)
        ~~~~^^^^^^^^^^^^^^^^^
      File "<doctest sage.doctest.test[25]>", line 1, in <module>
        pid = int(open(F).read())
    ValueError: invalid literal for int() with base 10: ''
**********************************************************************
File "src/sage/src/sage/doctest/test.py", line 261, in sage.doctest.test
Failed example:
    os.kill(pid, signal.SIGQUIT)  # 2 seconds passed => still alive
Exception raised:
    Traceback (most recent call last):
      File "/usr/lib/python3.13/site-packages/sage/doctest/forker.py", line 730, in _run
        self.compile_and_execute(example, compiler, test.globs)
        ~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      File "/usr/lib/python3.13/site-packages/sage/doctest/forker.py", line 1154, in compile_and_execute
        exec(compiled, globs)
        ~~~~^^^^^^^^^^^^^^^^^
      File "<doctest sage.doctest.test[27]>", line 1, in <module>
        os.kill(pid, signal.SIGQUIT)  # 2 seconds passed => still alive
                ^^^
    NameError: name 'pid' is not defined
**********************************************************************
File "src/sage/src/sage/doctest/test.py", line 263, in sage.doctest.test
Failed example:
    os.kill(pid, signal.SIGQUIT)  # 10 seconds passed => dead  # random
Expected:
    Traceback (most recent call last):
    ...
    ProcessLookupError: ...
Got:
    <BLANKLINE>
    Traceback (most recent call last):
      File "/usr/lib/python3.13/site-packages/sage/doctest/forker.py", line 730, in _run
        self.compile_and_execute(example, compiler, test.globs)
        ~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      File "/usr/lib/python3.13/site-packages/sage/doctest/forker.py", line 1154, in compile_and_execute
        exec(compiled, globs)
        ~~~~^^^^^^^^^^^^^^^^^
      File "<doctest sage.doctest.test[29]>", line 1, in <module>
        os.kill(pid, signal.SIGQUIT)  # 10 seconds passed => dead  # random
                ^^^
    NameError: name 'pid' is not defined
**********************************************************************
1 item had failures:
   4 of  56 in sage.doctest.test
    [55 tests, 4 failures, 175.01s wall]
----------------------------------------------------------------------
sage -t --long --warn-long 30.0 --random-seed=150612886400060636501286193160487656764 src/sage/src/sage/doctest/test.py  # 4 doctests failed
----------------------------------------------------------------------
```

--- a/src/sage/doctest/test.py
+++ b/src/sage/doctest/test.py
@@ -25,7 +25,7 @@ Unset :envvar:`TERM` when running doctests, see :issue:`14370`::
 
 Check that :issue:`2235` has been fixed::
 
-    sage: subprocess.call(["python", "-m", "sage.doctest", "--warn-long", "0",    # long time
+    sage: subprocess.call(["python3", "-m", "sage.doctest", "--warn-long", "0",    # long time
     ....:       "--random-seed=0", "--optional=sage", "longtime.rst"], **kwds)
     Running doctests...
     Doctesting 1 file.
@@ -36,7 +36,7 @@ Check that :issue:`2235` has been fixed::
     ----------------------------------------------------------------------
     ...
     0
-    sage: subprocess.call(["python", "-m", "sage.doctest", "--warn-long", "0",    # long time
+    sage: subprocess.call(["python3", "-m", "sage.doctest", "--warn-long", "0",    # long time
     ....:      "--random-seed=0", "--optional=sage", "-l", "longtime.rst"], **kwds)
     Running doctests...
     Doctesting 1 file.
@@ -50,7 +50,7 @@ Check that :issue:`2235` has been fixed::
 
 Check handling of tolerances::
 
-    sage: subprocess.call(["python", "-m", "sage.doctest", "--warn-long", "0",    # long time
+    sage: subprocess.call(["python3", "-m", "sage.doctest", "--warn-long", "0",    # long time
     ....:      "--random-seed=0", "--optional=sage", "tolerance.rst"], **kwds)
     Running doctests...
     Doctesting 1 file.
@@ -120,7 +120,7 @@ Check handling of tolerances::
 
 Test the ``--initial`` option::
 
-    sage: subprocess.call(["python", "-m", "sage.doctest", "--warn-long", "0",    # long time
+    sage: subprocess.call(["python3", "-m", "sage.doctest", "--warn-long", "0",    # long time
     ....:      "--random-seed=0", "--optional=sage", "-i", "initial.rst"], **kwds)
     Running doctests...
     Doctesting 1 file.
@@ -151,7 +151,7 @@ Test the ``--initial`` option::
 
 Test the ``--exitfirst`` option::
 
-    sage: subprocess.call(["python", "-m", "sage.doctest", "--warn-long", "0",    # long time
+    sage: subprocess.call(["python3", "-m", "sage.doctest", "--warn-long", "0",    # long time
     ....:      "--random-seed=0", "--optional=sage", "--exitfirst", "initial.rst"], **kwds)
     Running doctests...
     Doctesting 1 file.
@@ -179,7 +179,7 @@ manner (:issue:`26912`)::
     sage: from copy import deepcopy
     sage: kwds2 = deepcopy(kwds)
     sage: kwds2['env'].update({'SAGE_TIMEOUT': '1', 'CYSIGNALS_CRASH_NDEBUG': '1'})
-    sage: subprocess.call(["python", "-m", "sage.doctest", "--warn-long", "0",    # long time
+    sage: subprocess.call(["python3", "-m", "sage.doctest", "--warn-long", "0",    # long time
     ....:      "--random-seed=0", "--optional=sage", "99seconds.rst"], **kwds2)
     Running doctests...
     Doctesting 1 file.
@@ -196,7 +196,7 @@ manner (:issue:`26912`)::
 
 Test handling of ``KeyboardInterrupt`` in doctests::
 
-    sage: subprocess.call(["python", "-m", "sage.doctest", "--warn-long", "0",    # long time
+    sage: subprocess.call(["python3", "-m", "sage.doctest", "--warn-long", "0",    # long time
     ....:      "--random-seed=0", "--optional=sage", "keyboardinterrupt.rst"], **kwds)
     Running doctests...
     Doctesting 1 file.
@@ -219,7 +219,7 @@ Test handling of ``KeyboardInterrupt`` in doctests::
 
 Interrupt the doctester::
 
-    sage: subprocess.call(["python", "-m", "sage.doctest", "--warn-long", "0",    # long time
+    sage: subprocess.call(["python3", "-m", "sage.doctest", "--warn-long", "0",    # long time
     ....:      "--random-seed=0", "--optional=sage", "interrupt.rst"], **kwds)
     Running doctests...
     Doctesting 1 file.
@@ -238,7 +238,7 @@ doesn't hurt::
     sage: from copy import deepcopy
     sage: kwds2 = deepcopy(kwds)
     sage: kwds2['env']['DOCTEST_TEST_PID_FILE'] = F  # Doctester will write its PID in this file
-    sage: subprocess.call(["sage", "-tp", "1000000", "--timeout=120",  # long time
+    sage: subprocess.call(["python3", "-m", "sage.doctest", "-p", "1000000", "--timeout=120",  # long time
     ....:      "--die_timeout=10", "--optional=sage",
     ....:      "--warn-long", "0", "99seconds.rst", "interrupt_diehard.rst"], **kwds2)
     Running doctests...
@@ -274,7 +274,7 @@ random`` tag.
 
 Test a doctest failing with ``abort()``::
 
-    sage: subprocess.call(["python", "-m", "sage.doctest", "--warn-long", "0",    # long time
+    sage: subprocess.call(["python3", "-m", "sage.doctest", "--warn-long", "0",    # long time
     ....:      "--random-seed=0", "--optional=sage", "abort.rst"], **kwds)
     Running doctests...
     Doctesting 1 file.
@@ -300,7 +300,7 @@ A different kind of crash (also test printing of line continuation ``...:``,
 represented by ``<DOTSCOLON>`` below)::
 
     sage: # long time
-    sage: proc = subprocess.run(["python", "-m", "sage.doctest", "--warn-long", "0",
+    sage: proc = subprocess.run(["python3", "-m", "sage.doctest", "--warn-long", "0",
     ....:      "--random-seed=0", "--optional=sage", "fail_and_die.rst"], **kwds,
     ....:      stdout=subprocess.PIPE, text=True)
     sage: # the replacements are needed to avoid the strings being interpreted
@@ -336,7 +336,7 @@ represented by ``<DOTSCOLON>`` below)::
 
 Test that ``sig_on_count`` is checked correctly::
 
-    sage: subprocess.call(["python", "-m", "sage.doctest", "--warn-long", "0",    # long time
+    sage: subprocess.call(["python3", "-m", "sage.doctest", "--warn-long", "0",    # long time
     ....:      "--random-seed=0", "--optional=sage", "sig_on.rst"], **kwds)
     Running doctests...
     Doctesting 1 file.
@@ -362,7 +362,7 @@ Test that ``sig_on_count`` is checked correctly::
 Test logfiles in serial and parallel mode (see :issue:`19271`)::
 
     sage: t = tmp_filename()
-    sage: subprocess.call(["python", "-m", "sage.doctest", "--serial", "--warn-long", "0",    # long time
+    sage: subprocess.call(["python3", "-m", "sage.doctest", "--serial", "--warn-long", "0",    # long time
     ....:      "--random-seed=0", "--optional=sage", "--logfile", t, "simple_failure.rst"],
     ....:      stdout=open(os.devnull, "w"), **kwds)
     1
@@ -387,7 +387,7 @@ Test logfiles in serial and parallel mode (see :issue:`19271`)::
     ----------------------------------------------------------------------
     ...
 
-    sage: subprocess.call(["python", "-m", "sage.doctest", "--warn-long", "0",    # long time
+    sage: subprocess.call(["python3", "-m", "sage.doctest", "--warn-long", "0",    # long time
     ....:      "--random-seed=0", "--optional=sage", "--logfile", t, "simple_failure.rst"],
     ....:      stdout=open(os.devnull, "w"), **kwds)
     1
@@ -414,7 +414,7 @@ Test logfiles in serial and parallel mode (see :issue:`19271`)::
 
 Test the ``--debug`` option::
 
-    sage: subprocess.call(["python", "-m", "sage.doctest", "--warn-long", "0",    # long time
+    sage: subprocess.call(["python3", "-m", "sage.doctest", "--warn-long", "0",    # long time
     ....:      "--random-seed=0", "--optional=sage", "--debug", "simple_failure.rst"],
     ....:      stdin=open(os.devnull), **kwds)
     Running doctests...
@@ -449,7 +449,7 @@ Test the ``--debug`` option::
 
 Test running under gdb, without and with a timeout::
 
-    sage: subprocess.call(["python", "-m", "sage.doctest",  "--warn-long", "0",   # long time, optional: gdb
+    sage: subprocess.call(["python3", "-m", "sage.doctest",  "--warn-long", "0",   # long time, optional: gdb
     ....:      "--random-seed=0", "--optional=sage", "--gdb", "1second.rst"],
     ....:      stdin=open(os.devnull), **kwds)
     exec gdb ...
@@ -465,7 +465,7 @@ Test running under gdb, without and with a timeout::
 
 gdb might need a long time to start up, so we allow 30 seconds::
 
-    sage: subprocess.call(["python", "-m", "sage.doctest",  "--warn-long", "0",   # long time, optional: gdb
+    sage: subprocess.call(["python3", "-m", "sage.doctest",  "--warn-long", "0",   # long time, optional: gdb
     ....:      "--random-seed=0", "--optional=sage", "--gdb", "-T30", "99seconds.rst"],
     ....:      stdin=open(os.devnull), **kwds)
     exec gdb ...
@@ -475,7 +475,7 @@ gdb might need a long time to start up, so we allow 30 seconds::
 
 Test the ``--show-skipped`` option::
 
-    sage: subprocess.call(["python", "-m", "sage.doctest", "--warn-long", "0",    # long time
+    sage: subprocess.call(["python3", "-m", "sage.doctest", "--warn-long", "0",    # long time
     ....:      "--random-seed=0", "--optional=sage", "--show-skipped", "show_skipped.rst"], **kwds)
     Running doctests ...
     Doctesting 1 file.
@@ -494,7 +494,7 @@ Test the ``--show-skipped`` option::
 
 Optional tests are run correctly::
 
-    sage: subprocess.call(["python", "-m", "sage.doctest", "--warn-long", "0", "--long",  # long time
+    sage: subprocess.call(["python3", "-m", "sage.doctest", "--warn-long", "0", "--long",  # long time
     ....:      "--random-seed=0", "--show-skipped", "--optional=sage,gap", "show_skipped.rst"], **kwds)
     Running doctests ...
     Doctesting 1 file.
@@ -509,7 +509,7 @@ Optional tests are run correctly::
     ...
     0
 
-    sage: subprocess.call(["python", "-m", "sage.doctest", "--warn-long", "0", "--long",  # long time
+    sage: subprocess.call(["python3", "-m", "sage.doctest", "--warn-long", "0", "--long",  # long time
     ....:      "--random-seed=0", "--show-skipped", "--optional=gAp", "show_skipped.rst"], **kwds)
     Running doctests ...
     Doctesting 1 file.
@@ -527,7 +527,7 @@ Optional tests are run correctly::
 
 Test an invalid value for ``--optional``::
 
-    sage: subprocess.call(["python", "-m", "sage.doctest", "--warn-long", "0",
+    sage: subprocess.call(["python3", "-m", "sage.doctest", "--warn-long", "0",
     ....:      "--random-seed=0", "--optional=bad-option", "show_skipped.rst"], **kwds)
     Traceback (most recent call last):
     ...
@@ -542,7 +542,7 @@ Test ``atexit`` support in the doctesting framework::
     sage: from copy import deepcopy
     sage: kwds2 = deepcopy(kwds)
     sage: kwds2['env']['DOCTEST_DELETE_FILE'] = F
-    sage: subprocess.call(["python", "-m", "sage.doctest", "--warn-long", "0",    # long time
+    sage: subprocess.call(["python3", "-m", "sage.doctest", "--warn-long", "0",    # long time
     ....:      "--random-seed=0", "--optional=sage", "atexit.rst"], **kwds2)
     Running doctests...
     Doctesting 1 file.
@@ -562,7 +562,7 @@ Test ``atexit`` support in the doctesting framework::
 
 Test that random tests are reproducible::
 
-    sage: subprocess.call(["python", "-m", "sage.doctest", "--warn-long", "0",    # long time
+    sage: subprocess.call(["python3", "-m", "sage.doctest", "--warn-long", "0",    # long time
     ....:      "--random-seed=0", "--optional=sage", "random_seed.rst"], **kwds)
     Running doctests...
     Doctesting 1 file.
@@ -584,7 +584,7 @@ Test that random tests are reproducible::
     ----------------------------------------------------------------------
     ...
     1
-    sage: subprocess.call(["python", "-m", "sage.doctest", "--warn-long", "0",    # long time
+    sage: subprocess.call(["python3", "-m", "sage.doctest", "--warn-long", "0",    # long time
     ....:      "--random-seed=1", "--optional=sage", "random_seed.rst"], **kwds)
     Running doctests...
     Doctesting 1 file.

